Backup E2E test shows successful result even when failing

**Description**

Backup E2E test shows successful result even when failing. See for details: https://status.build.kyma-project.io/view/gcs/kyma-prow-logs/logs/kyma-gke-end-to-end-test-backup-restore/1102796075275653120

Log Line 533:

```
--- FAIL: TestBackupAndRestoreCluster (52.22s)
```

Log Line 719:

```# Job is finished (exit status: 1)
```

**Expected result**

Job should be marked as failed.

**Actual result**

Job is marked as successfull

**Steps to reproduce**

Run test again.

**Troubleshooting**

Looking at the logs.

--- a/installation/scripts/e2e-testing.sh
+++ b/installation/scripts/e2e-testing.sh
@@ -1,7 +1,4 @@
 #!/usr/bin/env bash
-
-set -x
-
 ROOT_PATH=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 source ${ROOT_PATH}/utils.sh
 source ${ROOT_PATH}/testing-common.sh
@@ -31,7 +28,7 @@ echo "----------------------------"
 echo "- E2E Testing Kyma..."
 echo "----------------------------"
 
-result=0
+exitCode=0
 
 for testcase in $(ls -d ${ROOT_PATH}/../../tests/end-to-end/*/deploy/chart/*)
 do
@@ -50,7 +47,7 @@ do
     then
         releasesToClean="$releasesToClean $release"
     else
-        result=$testResult
+        exitCode=$testResult
     fi
 done
 
@@ -58,7 +55,7 @@ checkAndCleanupTest end-to-end
 cleanupResult=$?
 if [ $cleanupResult -ne 0 ]
 then
-    result=$cleanupResult
+   exitCode=$cleanupResult
 fi
 
 
@@ -67,11 +64,11 @@ for release in $releasesToClean; do
     cleanupResult=$?
     if [ $cleanupResult -ne 0 ]
     then
-        result=$cleanupResult
+        exitCode=$cleanupResult
     fi
 done
 
-if [ ${result} -ne 0 ]
+if [ ${exitCode} -ne 0 ]
 then
     log FAIL red
     exit 1

--- a/installation/scripts/e2e-testing.sh
+++ b/installation/scripts/e2e-testing.sh
@@ -1,14 +1,20 @@
 #!/usr/bin/env bash
 ROOT_PATH=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
-source ${ROOT_PATH}/utils.sh
-source ${ROOT_PATH}/testing-common.sh
+source "${ROOT_PATH}/utils.sh"
+source "${ROOT_PATH}/testing-common.sh"
 
 
 cleanupHelmE2ERelease () {
     local release=$1
     log 'Running cleanup'
-    helm del --purge  $release
-    while helm list --deleting 2>/dev/null | grep $release ; do
+    helm del --purge "${release}"
+    local deleteErr=$?
+    if [ ${deleteErr} -ne 0 ]
+    then
+      log "FAILED cleaning release.\n" red
+      return 1
+    fi
+    while helm list --deleting 2>/dev/null | grep "${release}" ; do
         sleep 1
         echo .
     done
@@ -22,7 +28,10 @@ echo "-------------------------------"
 cleanupHelmTestPods end-to-end
 cleanupE2EErr=$?
 
-
+if [ ${cleanupE2EErr} -ne 0 ]
+then
+    exit 1
+fi
 
 echo "----------------------------"
 echo "- E2E Testing Kyma..."
@@ -30,26 +39,20 @@ echo "----------------------------"
 
 exitCode=0
 
-for testcase in $(ls -d ${ROOT_PATH}/../../tests/end-to-end/*/deploy/chart/*)
-do
-    release=$(basename $testcase)
-    cleanupHelmE2ERelease $release
 
-    if [ ${cleanupE2EErr} -ne 0 ]
-    then
-        exit 1
-    fi
+testcase="${ROOT_PATH}"/../../tests/end-to-end/backup-restore-test/deploy/chart/backup-test
+release=$(basename "$testcase")
 
-    helm install $testcase --name $release --namespace end-to-end
-    helm test $release --timeout 10000
-    testResult=$?
-    if [ $testResult -eq 0 ]
-    then
-        releasesToClean="$releasesToClean $release"
-    else
-        exitCode=$testResult
-    fi
-done
+cleanupHelmE2ERelease "${release}"
+helm install "$testcase" --name "${release}" --namespace end-to-end
+helm test "${release}" --timeout 10000
+testResult=$?
+if [ $testResult -eq 0 ]
+then
+    releasesToClean="$releasesToClean ${release}"
+else
+    exitCode=$testResult
+fi
 
 checkAndCleanupTest end-to-end
 cleanupResult=$?
@@ -60,7 +63,7 @@ fi
 
 
 for release in $releasesToClean; do
-    cleanupHelmE2ERelease $release
+    cleanupHelmE2ERelease "${release}"
     cleanupResult=$?
     if [ $cleanupResult -ne 0 ]
     then

